public/app/features/dashboard/components/SaveDashboard/SaveDashboardDrawer.tsx

```diff
@@ -0,0 +1,279 @@
+    return result.dashboard;
+  }, [dashboard, status]);
+
+  const data = useMemo(() => {
+    const clone = dashboard.getSaveModelClone({
+      saveTimerange: Boolean(saveTimeRange),
+      saveVariables: Boolean(saveVariables),
+    });
+
+    if (!previous.value) {
+      return { clone };
+    }
+
+    const cloneJSON = JSON.stringify(clone, null, 2);
+    const cloneSafe = JSON.parse(cloneJSON); // avoids undefined issues
+
+    const diff = jsonDiff(previous.value, cloneSafe);
+    let diffCount = 0;
+    for (const d of Object.values(diff)) {
+      diffCount += d.length;
+    }
+
+    return {
+      clone,
+      cloneJSON, // for diff
+      diff,
+      diffCount,
+      hasChanges: diffCount > 0,
+    };
+  }, [dashboard, previous.value, saveTimeRange, saveVariables]);
+
+  const [showDiff, setShowDiff] = useState(false);
+  const [saving, setSaving] = useState(false);
+
+  // reuse current save path
+  const { state, onDashboardSave } = useDashboardSave(dashboard);
+  const doSave = async (dto: FormDTO) => {
+    setSaving(true);
+
+    const body = data.clone;
+    if (status.isNew && dto.newDashboardTitle) {
+      body.title = dto.newDashboardTitle;
+    }
+
+    const result = await onDashboardSave(
+      body,
+      {
+        message: dto.message,
+        saveTimerange: saveTimeRange,
+        saveVariables: saveVariables,
+      },
+      dashboard
+    );
+
+    if (result.status === 'success') {
+      if (saveVariables) {
+        dashboard.resetOriginalVariables();
+      }
+      if (saveTimeRange) {
+        dashboard.resetOriginalTime();
+      }
+    }
+    onDismiss();
+    setSaving(false);
+  };
+
+  const renderBody = () => {
+    if (showDiff) {
+      return (
+        <>
+          {previous.loading && <Spinner />}
+          {!data.hasChanges && <div>No changes made to this dashboard</div>}
+          {data.diff && data.hasChanges && (
+            <div>
+              <div className={styles.spacer}>
+                {Object.entries(data.diff).map(([key, diffs]) => (
+                  <DiffGroup diffs={diffs} key={key} title={key} />
+                ))}
+              </div>
+
+              <h4>JSON Diff</h4>
+              <DiffViewer oldValue={JSON.stringify(previous.value, null, 2)} newValue={data.cloneJSON!} />
+            </div>
+          )}
+        </>
+      );
+    }
+
+    if (status.isNew) {
+      return (
+        <SaveDashboardAsForm
+          dashboard={dashboard}
+          onCancel={onDismiss}
+          onSuccess={onDismiss}
+          onSubmit={(clone, options, dashboard) => {
+            return onDashboardSave(clone, options, dashboard);
+          }}
+          isNew={status.isNew}
+        />
+      );
+    }
+
+    if (status.isProvisioned) {
+      return <SaveProvisionedDashboardForm dashboard={dashboard} onCancel={onDismiss} onSuccess={onDismiss} />;
+    }
+
+    return (
+      <form onSubmit={handleSubmit(doSave)}>
```

Comment:
This could probably be broken out into a separate component to reduce size / complexity of this file a bit.

Though given this is a WIP we probably can kick refactor / optimizations down the road a bit as this structure may change during refinement anyways